recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.tmpl

```diff
@@ -356,6 +351,9 @@ class EnvFirmware(Firmware):
             if self.env_mtd_num and self.env_bk_mtd_num:
                 break
 
+            if mtd_num > 20:
+                raise UpgradeError("EnvFirmware: No env partition found")
+
```

Comment:
I would use `for ... else ...`, to make it more pythonic :)

Reply:
Also add a comment to explain 20 is enough for now.

Reply:
Or is it better to wrap this logic into mtd_device.get_partition_by_label("env")?

Reply:
Done. However, given the development and testing effort involved, it's advisable to avoid making significant modifications to the original logic.